src/core/types/base/mempool.ts

```diff
@@ -98,17 +107,50 @@ export function decodeMempool(mempool: Mempool, ignoreAddresses: Record<string,
  * @param ignoreAddresses: Array containing wallet addresses that should be ignored and filtered.
  * @returns Boolean stating if `msg` is to be processed or not.
  */
-function isAllowedMempoolMsg(msg: MsgExecuteContract, ignoreAddresses: Record<string, boolean>): boolean {
+function isAllowedMempoolMsg(
+	msg: MsgExecuteContract,
+	ignoreAddresses: IgnoredAddresses,
+	timeoutDur: number,
+	iteration: number,
+): boolean {
 	//if the sender of the message is in our ignore list: skip this message
 	if (ignoreAddresses[msg.sender]) {
-		return false;
+		return !keepIgnored(msg.sender, undefined, ignoreAddresses, timeoutDur, iteration);
 		// if they use a contract to fund new wallets
 	} else if (ignoreAddresses[msg.contract]) {
 		const containedMsg = JSON.parse(fromUtf8(msg.msg));
-		const gets = fromAscii(fromBase64(containedMsg.delegate.msg));
-		ignoreAddresses[gets] = true;
+		if (containedMsg.delegate) {
+			const gets = fromAscii(fromBase64(containedMsg.delegate.msg));
+			return !keepIgnored(msg.sender, gets, ignoreAddresses, timeoutDur, iteration);
+		}
+	}
+	return true;
+}
+/**
+ *
+ */
+function keepIgnored(
+	address: string,
+	reciever: string | undefined,
+	ignoreAddresses: IgnoredAddresses,
+	timeoutDur: number,
+	iteration: number,
+) {
+	if (
+		ignoreAddresses[address].source ||
+		ignoreAddresses[address].timeout_at + ignoreAddresses[address].duration <= iteration
+	) {
+		if (reciever) {
+			ignoreAddresses[reciever] = {
+				source: false,
```

Comment:
Don't really understand why we need the "source"?

Reply:
you are right can check for timeoutAt == 0